server/routes/visitJourney/checkYourBooking.ts

```diff
@@ -39,12 +41,23 @@ export default class CheckYourBooking {
     const { prisonId } = req.session.selectedEstablishment
     const { offenderNo } = visitSessionData.prisoner
 
+    const urlPrefix = getUrlPrefix(isUpdate, visitSessionData.visitReference)
+
+    let { allowOverBooking } = req.session
+
+    if (req.body.confirmOverbooking === 'yes') {
+      allowOverBooking = true
+    } else if (req.body.confirmOverbooking === 'no') {
+      return res.redirect(`${urlPrefix}/select-date-and-time`)
+    }
+
```

Comment:
I think I understand what's going on here but I think it could be a bit simpler/clearer. E.g., something like

```
const { confirmOverBooking } = req.body  // this will be set if we have come from overbooking confirmation page
if (confirmOverBooking === 'no' ) {
  return res.redirect(`${urlPrefix}/select-date-and-time`)  // i.e. return early if we're going to
}

if (confirmOverBooking === 'yes' ) {
  visitSessionData.allowOverBooking = true
}

// ... and then further down where you call bookVisit(), just pass 'allowOverBooking: visitSessionData.allowOverBooking'
```